pkg/gofr/subscriber_test.go

```diff
@@ -102,7 +111,7 @@ func TestSubscriptionManager_SubscribeError(t *testing.T) {
 
 		// Run the subscriber in a goroutine
 		go func() {
-			subscriptionManager.startSubscriber("abc",
+			_ = subscriptionManager.startSubscriber(context.Background(), "abc",
```

Comment:
Shouldn't you return the error and test it here with assert.NoError?

Reply:
Since the test method is `TestSubscriptionManager_SubscribeError `, i have added assert.Contains() for error checking as `assert.ErrorIs` was resulting in failure:

```go
	expected: "subscription error: subscription error"
        	            	in chain: "subscription error: subscription error"
        	            		"subscription error"
```